go/vt/vttablet/tabletserver/schema/load_table.go

```diff
@@ -157,3 +171,39 @@ func getNum(in map[string]string, key string) (int, error) {
 	}
 	return v, nil
 }
+
+func getCols(in map[string]string, key string) ([]string, error) {
+	sv := in[key]
+	if sv == "" {
+		return nil, fmt.Errorf("attribute %s not specified for message table", key)
+	}
+	cols := strings.Split(sv, ";")
+	return cols, nil
+}
+
+func getDefaultMessageFields(tableFields []*querypb.Field, hiddenCols map[string]struct{}) []*querypb.Field {
+	fields := make([]*querypb.Field, 0, len(tableFields))
+	// Load user-defined columns. Any "unrecognized" column is user-defined.
+	for _, field := range tableFields {
+		if _, ok := hiddenCols[strings.ToLower(field.Name)]; ok {
+			continue
+		}
+
+		fields = append(fields, field)
+	}
+	return fields
+}
+
+func getAllowedMessageFields(tableFields []*querypb.Field, streamCols []string) []*querypb.Field {
+	fields := make([]*querypb.Field, 0, len(streamCols))
+	// Load user-defined columns. Any "unrecognized" column is user-defined.
+	for _, col := range streamCols {
+		for _, field := range tableFields {
+			if field.Name == col {
```

Comment:
This is case and white space sensitive, which is fine, probably just worth documenting as most identifier comparisons in SQL are not (or dependent on the collation used). We could potentially use the new evalengine here to compare the identifiers in the same way MySQL would.

Reply:
To explain what I mean here, it would be something like this:
```
func getAllowedMessageFields(tableFields []*querypb.Field, streamCols []string) []*querypb.Field {
	fields := make([]*querypb.Field, 0, len(streamCols))
	// Load user-defined columns. Any "unrecognized" column is user-defined.
	for _, col := range streamCols {
		for _, field := range tableFields {
			if res, _ := evalengine.NullsafeCompare(sqltypes.NewVarChar(field.Name), sqltypes.NewVarChar(col), collations.Default()); res == 0 {
				fields = append(fields, field)
				break
			}
		}
	}
	return fields
}
```

Reply:
And if we wanted to deal with the user potentially having whitespace in the comment value:
```
if res, _ := evalengine.NullsafeCompare(sqltypes.NewVarChar(strings.TrimSpace(field.Name)), sqltypes.NewVarChar(strings.TrimSpace(col)), collations.Default()); res == 0 {
				fields = append(fields, field)
				break
			}
```

Reply:
Perfect, I'll go ahead and use that. I may clean up the other functions here in a secondary commit too

Reply:
Cool! To be clear, this is what I was thinking for the final function (ignoring the function name which is a separate issue):
```
func getAllowedMessageFields(tableFields []*querypb.Field, streamCols []string) []*querypb.Field {
	fields := make([]*querypb.Field, 0, len(streamCols))
	// Load user-defined columns. Any "unrecognized" column is user-defined.
	for _, col := range streamCols {
		for _, field := range tableFields {
			if res, _ := evalengine.NullsafeCompare(sqltypes.NewVarChar(field.Name),
				sqltypes.NewVarChar(strings.TrimSpace(col)),
				collations.Default()); res == 0 {
				fields = append(fields, field)
				break
			}
		}
	}
	return fields
}
```

This way the comparison is case insensitive as the default collation with MySQL 5.7 is `utf8mb4_general_ci` and with 8.0 is `utf8mb4_0900_ai_ci` and we're stripping any leading or trailing whitespace from the identifiers in the comment directive as I can see that happening (in a list you may expect whitespace not to matter).

Reply:
done